src/main/java/org/glassfish/enterprise/concurrent/ManagedThreadFactoryImpl.java

```diff
@@ -187,20 +187,18 @@ protected Collection<Thread> getThreads() {
         return result;
     }
     public void taskStarting(Thread t, ManagedFutureTask task) {
-        if (t instanceof ManagedThread) {
-            ManagedThread mt = (ManagedThread) t;
+        if (t instanceof ThreadWithTiming) {
```

Comment:
You can omit the cast and just write

`if (t instanceof ThreadWithTiming mt)`

Reply:
I was using still JDK 11, but you are right, pom was upgraded to 17. I updated all such cases in this file and ForkJoinManagedExecutorService.